docs/architecture/api-20201231-preview.md

```diff
@@ -0,0 +1,133 @@
+# API 20201231-preview design
+
+For api `20201231-preview` bellow is the list of proposed changes and features.
+These would have to land into API and backend as a features.
+
+Feature work will need to be agreed and converted to working stories. All technical details bellow are overview only
+
+## API changes
+
+### .OpenShiftClusterProperties.SecurityProfile
+
+Add security profile to worker pool configuration for all security enhancements
+for the product compute resource.
+
+We need to make sure that these option are validated on the cluster too when customer is interacting with MachineSet objects. We should be able to verify
+if these options where enabled on cluster create and set those accordingly.
+
+```
+	// MasterProfile represents a master profile.
+
+type MasterProfile struct {
+	...
+
+	ComputeSecurityProfile ComputeSecurityProfile `json:"securityProfile,omitempty"`
+}
+
+
+// WorkerProfile represents a worker profile.
+type WorkerProfile struct {
+	...
+
+	ComputeSecurityProfile ComputeSecurityProfile `json:"securityProfile,omitempty"`
+}
+```
+
+Re-use same structure for all compute profiles:
+```
+// ComputeSecurityProfile represents an security profile for all compute
+type ComputeSecurityProfile struct {
+	// EncryptionAtHost value sets encryptionAtHost option for all VirtualMachines.
+	EncryptionAtHost *bool `json:"encryptionAtHost,omitempty"`
```

Comment:
Add BYOK flag

Reply:
Boolean add enum as per rest api spec

Reply:
Does `ComputeSecurityProfie` map to the Azure [`compute.SecurityProfile`](https://godoc.org/github.com/Azure/azure-sdk-for-go/services/compute/mgmt/2020-06-30/compute#SecurityProfile)? If it does and since our API will be published as part of the official Azure SDK, I suggest using the same name, to communicate their relationship.

Reply:
my thinking was similar hence the name ComputeSecurityProfile (SecurityProfile is a bit ambiguous).

Reply:
What's the package name? If it's `compute`, then `compute.SecurityProfile` won't be ambigiuous and align with the [go code review wiki](https://github.com/golang/go/wiki/CodeReviewComments#package-names).